python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1160,6 +1160,10 @@ class Cast(Expr):
     def __init__(self, dtype: plc.DataType, value: Expr) -> None:
         super().__init__(dtype)
         self.children = (value,)
+        if not plc.unary.is_supported_cast(self.dtype, value.dtype):
+            raise NotImplementedError(
+                f"Can't cast {self.dtype.id().name} to {value.dtype.id().name}"
+            )
```

Comment:
Test this branch please.

Reply:
https://github.com/rapidsai/cudf/pull/16232/commits/a2858e6fd919732c5eff7349a6fe59efe127c2fd should cover things